### PR TITLE
[WOR-1372] When cloning a requester pays workspace, show a generic egress message.

### PR DESCRIPTION
**Pull request by cahrens**
Ticket: https://broadworkbench.atlassian.net/browse/WOR-1372

When going to a requester pays workspace, the bucket location cannot be fetched. We currently show this in the bucket location section:

image

However, trying to bring up the clone workspace dialog would throw an error because we attempt to get the bucket location so that we can display an egress warning:

(Current situation on dev/prod):
image

To fix this, we catch the requester pays error and show a generic egress message when the user selects a billing project to clone to… in this case, we don't know if the user will actually hit an egress issue.

Requester pays workspace:
image

Contrast with the case of a non-requester pays workspace, where we only show the warning when the user WILL incur egress:
image
image

### CODE REVIEW

File: src/pages/workspaces/workspace/Dashboard/BucketLocation.ts [hunk 1]
```diff
@@ -42,7 +42,7 @@ export const BucketLocation = requesterPaysWrapper({ onDismiss: _.noop })((props
         .checkBucketLocation(googleProject, bucketName);
       setBucketLocation(response);
     } catch (error) {
-      if (error != null && typeof error === 'object' && 'requesterPaysError' in error) {
+      if (isBucketErrorRequesterPays(error)) {
```
Comment (cahrens) [inline src/pages/workspaces/workspace/Dashboard/BucketLocation.ts:45]: This was actually a bug in that we didn't check that `error.requesteredPaysError` was true… we just checked that the key existed in the object. If the error thrown is not caused by requester pays, `error.requesterPaysError` will be false. See https://github.com/DataBiosphere/terra-ui/blob/692dbfd07933043f7d13e0ef309d6a81883f200b/src/libs/ajax/ajax-common.ts#L147, which is also modified in this PR to enforce typing.

File: src/workspaces/NewWorkspaceModal/NewWorkspaceModal.test.ts [hunk 2]
```diff
@@ -1396,4 +1401,103 @@ describe('NewWorkspaceModal', () => {
       })
     );
   });
+
+  describe('shows egress warnings for cloning GCP workspaces', () => {
+    it('shows a message if the destination bucket location is in a different region', async () => {
```
Comment (cahrens) [inline src/workspaces/NewWorkspaceModal/NewWorkspaceModal.test.ts:1406]: This is a test case for our existing functionality (which was written before we had much of anything in the way of unit tests).

File: src/workspaces/NewWorkspaceModal/NewWorkspaceModal.test.ts [hunk 3]
```diff
@@ -1396,4 +1401,103 @@ describe('NewWorkspaceModal', () => {
...
+
+      // Assert
+      // Have to use textContent to work around bolded sections of text.
+      const warning = screen.getByText('Copying data from', { exact: false });
+      expect(warning.textContent).toEqual(
+        'Copying data from us-central1 (Iowa) to northamerica-northeast1 (Montreal) may incur network egress charges. '
+      );
+    });
+
+    it('shows a generic message if the source workspace is requester pays', async () => {
```
Comment (cahrens) [inline src/workspaces/NewWorkspaceModal/NewWorkspaceModal.test.ts:1461]: Test case for the new handling of requester pays error.

**Review by marctalbott — APPROVED** [review]

**Comment by cahrens** [comment]
I noticed that in the "non-requester pays" case, we would show the egress warning before the billing project was selected if the source workspace had a bucket location that did not match the default bucket location. I thought this was confusing because the user isn't yet seeing the destination bucket location, so I pushed up another [commit](97b0fc6bb357f19fe999c9c196bc404de5a99a05) to make the same modification in that case that I did in the generic one-- don't show the egress message until a google project is selected.

Before:
image

After:
image

image

**Review by blakery — APPROVED** [review]

**Pull request merged**